ebean-core/src/main/java/io/ebeaninternal/server/query/LimitOffsetPagedList.java

```diff
@@ -85,22 +92,23 @@ public int getTotalPageCount() {
 
   @Override
   public int getTotalCount() {
+    // already fetched?
+    if (totalRowCount > -1) return totalRowCount;
```

Comment:
This needs to move inside the lock ... (or be volatile? but I think just move inside the lock)